language-design.md

```diff
@@ -28,7 +41,9 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
+Several features that identify a well-designed language are the following: easy to extend, comprehensible, usability, good documentation, non-bitter user base. Essentially, a well designed language won’t leave its users frustrated for hours on end as they struggle to program the basic ideas they have. In other words, since a programming language is a way of expression, a good language should be intuitive and self consistent. It will have documentation to support its design and the syntax will be readable (enough). Furthermore, as Steele highlights often in _Growing a Language_, it will have the ability to be extended and thus adapted by its user base. This evolution allows users to grow the language in a direction that best suits the needs of the masses. An example of this, as explored by Mernik, shows how the COM interface allows Excel’s limited macro language to interface with languages such as C++, Java, and Basic through a “process of componentization… called automation [Chappell 1996]” [Mernik et al., 2005, 318]. By allowing a language, while staying specific to its domain, be more accessible to other interfaces, its use cases grow. This engenders a positive user base that speaks well of the language, rather than the stink eye from the computer science community.
```

Comment:
This is a really interesting idea, that you can tell how good a PL is by looking at its user base rather than the PL itself. It really speaks to the idea of user-centered design, which seems to be at the center of all of these readings.

I'm not totally sure how the COM interface fits into this - was it an interface that was created by users, or was it created by Excel designers to make the language more extensible? What is this process of automation, at a high level? I'd have liked some more explanation of this because it seems like it could be a useful example of what you're saying, but I haven't read this article by Mernik et al.